core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1458,22 +1476,7 @@ class Block : public lifecycle::StateMachine<Derived>, protected std::tuple<Argu
                 const auto data   = messageField<property_map>(message, gr::message::key::Data).value();
                 auto       notSet = settings().set(data);
 
-                std::string keysNotSet;
-                for (const auto &[k, v] : notSet) {
-                    keysNotSet += " " + k;
-                }
-
-                Message settingsUpdated;
-                settingsUpdated[gr::message::key::Kind] = gr::message::kind::SettingsChanged;
-                settingsUpdated[gr::message::key::Data] = settings().get();
-
-                if (!notSet.empty()) {
-                    Message errorMessage;
-                    errorMessage[gr::message::key::Kind]         = gr::message::kind::Error;
-                    errorMessage[gr::message::key::Data]         = notSet;
-                    settingsUpdated[gr::message::key::ErrorInfo] = std::move(errorMessage);
-                }
-                emitMessage(msgOut, std::move(settingsUpdated));
+                applyChangedSettings(notSet);
```

Comment:
The pattern of staging parameters in non-real-time and applying them from the real-time thread is a common pattern in embedded real-time computing.

As discussed in the chat `applyChangedSettings(...)` may not be called from `processMessages(...)` for thread-safety and real-time latency constraints reasons. 

The other uses that are invoked from the `work(...)` or `process[Bulk, One](...)` contexts are OK though. I.e. the processMessage function may add/stage new settings but they can only be applied from a real-time processing thread point of view. 

Thus the UI will and can reliably only get a delayed reply of when the settings are actually being applied.

In most cases (continuous processing) the delay should be rather small though (few ms upt to at most <~100 ms).